### ISSUE TITLE
Support for reflecting on enum types

### ISSUE THREAD

**Issue by chibisuke** [typescript-rtti/typescript-rtti#53]
Reproduction code:
```
import 'reflect-metadata';
import { reflect } from 'typescript-rtti';

enum test {
    t1,
    t2,
    t3
}

class TestCls {
    foo?: test;
}

console.log(reflect(TestCls).getOwnProperty('foo').type.toString());
```
output:
```
[0 | 1 | 2]
```

Expected result: 
reflect(TestCls).getOwnProperty('foo').type to be a instance of ReflectedClassRef

**Comment by rezonant** [comment]
Should be possible to reflect on enums at runtime unless the TS checker says otherwise. Will see if we can make this work!

**Comment by rezonant** [comment]
This is now available in 0.6.0.
